### PR TITLE
chore: replace harness with scenario (charm)

### PR DESCRIPTION
**Pull request by gruyaume**
# Description

This is the last of a series of PR in the Vault k8s charm to replace the harness ops testing framework with scenario. Here the focus is on the charm code itself.

# Checklist:

- [ ] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that validate the behaviour of the software
- [ ] I validated that new and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tests/unit/test_charm_configure.py [hunk 1]
```diff
@@ -0,0 +1,683 @@
...
+    #     relation_id=vault_kv_relation_id, remote_unit_name="vault-kv-remote/0"
+    # )
+    # event = Mock()
+    # event.relation_id = vault_kv_relation_id
+    # self.harness.charm._on_new_vault_kv_client_attached(event)
+    # patch_set_vault_url.assert_not_called()
+    # patch_set_mount.assert_not_called()
+    # patch_set_ca_certificate.assert_not_called()
+    # patch_audit_device_enabled.assert_not_called()
+
```
[the remark this answers is not in the document]
Comment (gruyaume) [reply tests/unit/test_charm_configure.py]: Done

File: tests/unit/test_charm_restore_backup_action.py [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+        action = scenario.Action(
+            name="restore-backup",
+        )
+
+        action_output = self.ctx.run_action(action, state_in)
+
+        assert action_output.success is False
+        assert action_output.failure == "Failed to restore vault."
+
+    def test_given_when_then(self):
```
[the remark this answers is not in the document]
Comment (gruyaume) [reply tests/unit/test_charm_restore_backup_action.py]: Oops, done

File: tests/unit/test_charm_list_backups_action.py [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+        )
+
+        action_output = self.ctx.run_action(action, state_in)
+
+        assert action_output.success is False
+        assert (
+            action_output.failure == "Failed to run list-backups action - Failed to list backups."
+        )
+
+    def test_given_s3_available_when_list_backups_then_backup_created(self):
```
[the remark this answers is not in the document]
Comment (gruyaume) [reply tests/unit/test_charm_list_backups_action.py]: Done

File: tests/unit/test_charm_create_backup_action.py [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+        action = scenario.Action(
+            name="create-backup",
+        )
+
+        action_output = self.ctx.run_action(action, state_in)
+
+        assert action_output.success is False
+        assert action_output.failure == "Failed to create S3 session."
+
+    def test_given_bucket_not_created_when_create_backup_then_action_fails(self):
```
[the remark this answers is not in the document]
Comment (gruyaume) [reply tests/unit/test_charm_create_backup_action.py]: Done

**Review by ghislainbourgeois — APPROVED** [review]

**Pull request merged**